### PR TITLE
RHEVDOCS3492, 3761: Aligned SBO 1.0.1 changes for IBM Z and Power

### PR DESCRIPTION
**Pull request by SNiemann15**
- OCP version for cherry-picking: enterprise-4.9, enterprise-4.10

- JIRA issues: [RHDEVDOCS-3492](https://issues.redhat.com/browse/RHDEVDOCS-3492), [RHDEVDOCS-3761](https://issues.redhat.com/browse/RHDEVDOCS-3761)
- Google doc: https://docs.google.com/document/d/1BtPM-JliJFY7EwnhbDEXu1bofmtp1OvTz4TXzMLjGds/edit
-  Related PR for x86: https://github.com/openshift/openshift-docs/pull/42505
- Preview pages: https://deploy-preview-42121--osdocs.netlify.app/openshift-enterprise/latest/applications/connecting_applications_to_services/getting-started-with-service-binding-ibm-power-ibm-z.html
- QE review: 
  - IBM Z - Johanna Husta
  - IBM Power - Pravin D'silva

### CODE REVIEW

File: modules/sbo-creating-a-postgresql-database-instance-power-z.adoc [hunk 1]
```diff
@@ -45,53 +34,31 @@ EOD
...
-nohup oc -n "\${cluster_namespace}" port-forward svc/sampledatabase 5432:5432 &
-sleep 5
-curl -LO https://raw.githubusercontent.com/spring-petclinic/spring-petclinic-rest/master/src/main/resources/db/postgresql/initDB.sql
-psql -h localhost -U "\${pgo_cluster_username}" "\${pgo_cluster_name}" -f initDB.sql
-curl -LO https://raw.githubusercontent.com/spring-petclinic/spring-petclinic-rest/master/src/main/resources/db/postgresql/populateDB.sql
-psql -h localhost -U "\${pgo_cluster_username}" "\${pgo_cluster_name}" -f populateDB.sql
-EOD
-----
-
-The output in the terminal shows you that the database is being configured for the application.
```
Comment (pravin-dsilva) [inline modules/sbo-creating-a-postgresql-database-instance-power-z.adoc:95]: Need this to populate the DB with the sample data.

**Comment by Srivaralakshmi** [comment]
@SNiemann15: PTAL at this comment in my PR: https://github.com/openshift/openshift-docs/pull/41809/#discussion_r811026249

I prefer us to maintain consistency in our QSG content and writing style. Hence sending my pov for your consideration.

File: modules/sbo-creating-a-postgresql-database-instance-power-z.adoc (hunk 1 above)
Comment (pmacik) [reply modules/sbo-creating-a-postgresql-database-instance-power-z.adoc:95]: with the new app, there is no need for populating the DB with the sample data, the application should initialize the DB o its own.

**Comment by Srivaralakshmi** [comment]
@SNiemann15 Just a small observation. The [QSG for x86](https://deploy-preview-41809--osdocs.netlify.app/openshift-enterprise/latest/applications/connecting_applications_to_services/getting-started-with-service-binding.html#sbo-creating-a-postgresql-database-instance_getting-started-with-service-binding) does not contain this prerequisite anymore: "You have installed PostgreSQL `psql` CLI." I did mention it in the [list of changes over slack](https://coreos.slack.com/archives/CHK22LM6Y/p1645112347537279?thread_ts=1641982374.031700&cid=CHK22LM6Y) explicitly (the 1st one in the list). PTAL and do the needful for the SBO QSG for IBM Z and Power.

**Comment by SNiemann15** [comment]
> @SNiemann15 Just a small observation. The [QSG for x86](https://deploy-preview-41809--osdocs.netlify.app/openshift-enterprise/latest/applications/connecting_applications_to_services/getting-started-with-service-binding.html#sbo-creating-a-postgresql-database-instance_getting-started-with-service-binding) does not contain this prerequisite anymore: "You have installed PostgreSQL `psql` CLI." I did mention it in the [list of changes over slack](https://coreos.slack.com/archives/CHK22LM6Y/p1645112347537279?thread_ts=1641982374.031700&cid=CHK22LM6Y) explicitly (the 1st one in the list). PTAL and do the needful for the SBO QSG for IBM Z and Power.

@Srivaralakshmi I know I left it there for the moment because @johusta wanted to verify if the prereq is still needed for P/Z or not. Once I get confirmation that it's not needed anymore I will remove.

**Comment by pmacik** [comment]
@johusta Looks like the dev4dev postgresql uses a different dialect for SQL, let me try to fix it - once I do, I'll re-push the updated image to the same URI `quay.io/service-binding/spring-petclinic:latest` so it should not affect the content of the docs

File: modules/sbo-creating-a-postgresql-database-instance-power-z.adoc [hunk 2]
```diff
@@ -6,27 +6,16 @@ To create a PostgreSQL database instance, you must create a `Database` custom re
...
-    type: postgresql
-    port: "5432"
-    service.binding/database: 'path={.spec.databaseName}'
-    service.binding/port: 'path={.metadata.annotations.port}'
-    service.binding/password: 'path={.spec.databasePassword}'
-    service.binding/username: 'path={.spec.databaseUser}'
-    service.binding/type: 'path={.metadata.annotations.type}'
-    service.binding/host: 'path={.metadata.annotations.host}'
-spec:
+ spec:
```
Comment (pmacik) [inline modules/sbo-creating-a-postgresql-database-instance-power-z.adoc:18]: this should not change - that introduces a wrong indentation and the apply fails

**Comment by pmacik** [comment]
@johusta Looks the old PG image version (9.6) might be the problem with the new app. Could you you try to update the DB image (https://github.com/openshift/openshift-docs/pull/42121/files#diff-807b32351555f711f63cdde21d3ef6702d04bed8c5af4b9f1b79f703129828b5R30) from `registry.redhat.io/rhel8/postgresql-96:latest`  (which is [deprecated](https://catalog.redhat.com/software/containers/rhel8/postgresql-96/5ba0ad1f5a134643ef2eeb9d) anyway ) to `registry.redhat.io/rhel8/postgresql-13:latest` ?

**Comment by johusta** [comment]
@SNiemann15 If we can get this new e2e example working then we will not need the psql cli. Let me know when you have updates for the indentation fixes and I can verify that. I have sent a request to the odo dev team to see if they can update the dev4devs operator to use the newer postgreSQL image.

**Comment by johusta** [comment]
Step 4. Setup port forwarding...... in the port forward string change spring-petclinic-rest to spring-petclinic

File: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc [hunk 3]
```diff
@@ -11,25 +12,24 @@ To connect the sample application to the database service, you must create a `Se
...
-    namespace: my-postgresql
+    name: spring-petclinic-pgcluster
 spec:
-    services: <1>
+  services: <1>
     - group: postgresql.dev4devs.com
       version: v1beta1
       kind: Database <2>
       name: sampledatabase
       version: v1alpha1
```
Comment (pravin-dsilva) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:27]: the correct version `v1alpha1` is already specified so `version: v1beta1` should be removed. @johusta  could you confirm that the version is `version: v1alpha1` on Z as well. Thanks.
```suggestion
    - group: postgresql.dev4devs.com
      kind: Database <2>
      name: sampledatabase
      version: v1alpha1
```

Comment (johusta) [reply modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:27]: Good catch Pravin I added comment to the google doc

**Comment by johusta** [comment]
Looks Good To Me

**Comment by pravin-dsilva** [comment]
lgtm

File: modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc [hunk 4]
```diff
@@ -41,77 +40,45 @@ The output verifies that the `ServiceBinding` CR is created to project the bindi
...
+spring-petclinic-postgresql   True    ApplicationsBound   47m
 ----
 +
-By default, the values from the binding data of the database service are projected as files into the workload container that runs the sample application.
+By default, the values from the binding data of the database service are projected as files into the workload container that runs the sample application. For example, all the values from the Secret resource are projected into the `bindings/spring-petclinic-pgcluster` directory.
 
 . Once this is created, you can go to the topology to see the visual connection.
 +
-.Connecting spring-petclinic-rest to sampledatabase
+.Connecting spring-petclinic to sample database
```
Comment (rolfedh) [inline modules/sbo-connecting-spring-petclinic-sample-app-to-postgresql-database-service-power-z.adoc:64]:
```suggestion
.Connecting spring-petclinic to a sample database
```

File: modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc [hunk 5]
```diff
@@ -62,55 +65,34 @@ The output verifies that the Spring PetClinic sample application is created and
...
-For example, if you try to access the list of all pets using `curl`, you can see an error message similar to the following sample message:
-+
-[source,text]
+[source,terminal]
 ----
-$ curl -X GET "http://localhost:9966/petclinic/api/pets" -H "accept: application/json"
+$ oc get pods -n my-petclinic
 ----
 +
+It will take a few minutes until the `CrashLoopBackOff` status is displayed:
```
Comment (rolfedh) [inline modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:87]:
```suggestion
It takes a few minutes until the `CrashLoopBackOff` status is displayed:
```

File: modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc [hunk 6]
```diff
@@ -62,55 +65,34 @@ The output verifies that the Spring PetClinic sample application is created and
...
 ----
-{"className":"org.springframework.transaction.CannotCreateTransactionException","exMessage":"Could
-not open JPA EntityManager for transaction; nested exception is
-org.hibernate.exception.JDBCConnectionException: Unable to acquire JDBC
-Connection"}
+NAME                                READY   STATUS             RESTARTS      AGE
+spring-petclinic-5b4c7999d4-wzdtz   0/1     CrashLoopBackOff   4 (13s ago)   2m25s
 ----
++
+At this stage, the pod fails to start. If you try to interact with the application, it will return errors.
```
Comment (rolfedh) [inline modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:96]:
```suggestion
At this stage, the pod fails to start. If you try to interact with the application, it returns errors.
```

**Review by rolfedh — APPROVED** [review]
Nice work. I made some minor suggestions.

File: modules/sbo-deploying-a-postgresql-database-operator-power-z.adoc [hunk 7]
```diff
@@ -64,11 +67,15 @@ EOD
 * For IBM Power: `quay.io/ibm/operator-registry-ppc64le:release-4.9` 
 * For IBM Z and LinuxONE: `quay.io/ibm/operator-registry-s390x:release-4.8`
 
-After the operator is installed, list the operator subscriptions in `openshift-operators` namespace:
+.Verification 
+
+. After the operator is installed, list the operator subscriptions in `openshift-operators` namespace:
```
Comment (Srivaralakshmi) [inline modules/sbo-deploying-a-postgresql-database-operator-power-z.adoc:72]:
```suggestion
. After the operator is installed, list the operator subscriptions in the `openshift-operators` namespace:
```

File: modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc [hunk 8]
```diff
@@ -62,55 +65,34 @@ The output verifies that the Spring PetClinic sample application is created and
...
-. Access link:http://localhost:9966/petclinic[http://localhost:9966/petclinic].
-+
-You can now remotely access the Spring PetClinic sample application at localhost:9966.
 +
 [NOTE]
 ====
-The application is not yet connected to the database service. If you try to interact with the application, it will return errors.
+If you are deploying the application using *Container images* in the *Developer* perspective of the web console, you must enter the following environment variables under the *Deployment* section of the *Advanced options*:
+* Name: SPRING_PROFILES_ACTIVE
+* Value: postgres
```
Comment (Srivaralakshmi) [inline modules/sbo-deploying-the-spring-petclinic-sample-application-power-z.adoc:76]: Suggestion: Formatting issue. please check and rectifiy.

**Review by Srivaralakshmi — CHANGES_REQUESTED** [review]
@SNiemann15 Nice work! Just left a couple of suggestions; nitpicks. PTAL. Thanks!

**Pull request merged**